pkgs/development/libraries/avahi/default.nix

```diff
@@ -33,6 +33,10 @@ stdenv.mkDerivation rec {
 
   patches = [
     ./no-mkdir-localstatedir.patch
+    (fetchurl {
```

Comment:
This is a generated patch so `fetchpatch` needs to be used.

Also it is better to use `https://github.com/lathiat/avahi/commit/9d31939e55280a733d930b15ac9e4dda4497680c.patch` URL, since it is easier to find the corresponding commit and view it on GitHub that way.